rcmd/install_test.go

```diff
@@ -39,49 +36,203 @@ func TestInstallArgs(t *testing.T) {
+//		installType  			string
+//		repoURL      			string
+//		repo         			string
+//		expectedRepo			string
+//		expectedOriginalRepo	string
+//		message      			string
+//	}{
+//		{
+//			filename:     "testsite/golden/simple/test-library/R6/Description",
+//			version:      "version",
+//			installType:  "binary",
+//			repoURL:      "myURL",
+//			repo:         "CRAN",
+//			expectedRepo: "CRAN",
+//			expectedOriginalRepo: "",
+//			message:      "R6 test",
+//		},
+//		{
+//			filename:     "testsite/golden/simple/test-library/pillar/Description",
+//			version:      "1.2.3",
+//			installType:  "binary",
+//			repoURL:      "www.myURL.com",
+//			repo:         "AlCran_Mandragoran",
+//			expectedRepo: "AlCran_Mandragoran",
+//			expectedOriginalRepo: "CRAN",
+//			message:      "pillar test",
+//		},
+//	}
+//
+//	for _, tt := range tests {
+//
+//		dcf, err := updateDescriptionInfo(tt.filename, tt.version, tt.installType, tt.repoURL, tt.repo)
+//
+//		var dcfBytes [][]byte
+//		for _, s := range dcf {
+//			dcfBytes = append(dcfBytes, []byte(s))
+//		}
+//
+//		installedPackage, _ := desc.ParseDesc((dcf))
+//
+//		assert.Equal(t, nil, err, fmt.Sprintf("Error: %s", err))
+//		assert.Equal(t, tt.expectedRepo, installedPackage.Repository, fmt.Sprintf("Failed: %s", tt.message))
+//		assert.Equal(t, tt.version, installedPackage.PkgrVersion, fmt.Sprintf("Failed: %s", tt.message))
+//		assert.Equal(t, tt.repoURL, installedPackage.PkgrRepositoryURL, fmt.Sprintf("Failed: %s", tt.message))
+//		assert.Equal(t, tt.installType, installedPackage.PkgrInstallType, fmt.Sprintf("Failed: %s", tt.message))
+//		assert.Equal(t, tt.expectedOriginalRepo, installedPackage.OriginalRepository, fmt.Sprintf("Failed: %s", tt.message))
+//	}
+//}
+
+func TestUpdateDescriptionInfoByLines_RepoUpdated(t *testing.T) {
+	tests := map[string]struct{
+		startingLines []string
+		version string
+		installType string
+		repoURL string
+		repo string
 	}{
-		{
-			filename:     "../integration_tests/simple/test-library/R6/Description",
-			version:      "version",
-			installType:  "binary",
-			repoURL:      "myURL",
-			repo:         "CRAN",
-			expectedRepo: "CRAN",
-			message:      "R6 test",
-		},
-		{
-			filename:     "../integration_tests/simple/test-library/pillar/Description",
-			version:      "1.2.3",
-			installType:  "binary",
-			repoURL:      "www.myURL.com",
-			repo:         "GitHub",
-			expectedRepo: "CRAN GitHub",
-			message:      "pillar test",
+		"Repository Upated": {
+			startingLines: []string{"Package: R6", "Version: 2.4.0", "Repository: CRAN"},
+			version: "pkgr0.0.test",
+			installType: "binary",
+			repoURL: "https://www.fakecranrepos.org",
+			repo: "AlCRAN_Mandragoran",
 		},
 	}
+	fmt.Println(tests["test1"])
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			results := updateDescriptionInfoByLines(test.startingLines, test.version, test.installType, test.repoURL, test.repo)
+			assert.Len(t, results, 7)
```

Comment:
so this actually breaks the entire table driven test abstraction. The idea is that the assertions in the loop should be applicable for each test - this is a "table" test with just one test, thus these pass. But what we should have a number of types of tests that flex the different scenarios

Reply:
I will not fix this for now since it shouldn't impact validation docs (as per our Slack discussion), but I understand the rationale and philosophy here and will write tests that way going forward.